src/contracts/ParaSwapDebtSwapAdapter.sol

```diff
@@ -87,38 +87,57 @@ abstract contract ParaSwapDebtSwapAdapter is
       debtSwapParams.debtAsset,
       debtSwapParams.debtRepayAmount,
       debtSwapParams.debtRateMode,
+      address(0), // debtSwapParams.newDebtAsset,
+      0, //debtSwapParams.maxNewDebtAmount,
       debtSwapParams.paraswapData,
       debtSwapParams.offset,
       msg.sender
     );
-    _flash(flashParams, debtSwapParams);
+
+    // If we need extra collateral, execute the flashloan with the collateral instead of the debt.
+    address assetToFlash;
+    uint256 amountToFlash;
+    if (debtSwapParams.extraCollateralAsset != address(0)) {
```

Comment:
just curious, codestyle wise i would have done this differently i think :thinking: 

Is it better to define local vars and have one combined call to ` _flash(flashParams, assetToFlash, amountToFlash);`

I would have done:
```
    if (debtSwapParams.extraCollateralAsset != address(0)) {
      flashParams.nestedFlashloanDebtAsset = debtSwapParams.newDebtAsset;
      flashParams.nestedFlashloanDebtAmount = debtSwapParams.maxNewDebtAmount;
      _flash(flashParams, debtSwapParams.extraCollateralAsset, debtSwapParams.extraCollateralAmount);
    }
   else {
         _flash(flashParams, debtSwapParams.newDebtAsset, debtSwapParams.maxNewDebtAmount);
   }
```
i think... for me feels cleaner, but can't really argue why. lmk what you think.

Reply:
I prefer your method as well, not really a fan of the local variable => assign in smaller scope => function call pattern cause it's more obfuscated when you actually execute the function call. Nice refactor!

Reply:
Addressed in [bf51e02](https://github.com/bgd-labs/aave-debt-swap/pull/17/commits/bf51e0214d1ac592833c11e8f1442cff77446d78)